### PR TITLE
Add RandomGenerator

### LINKED ISSUE #3: Prepare class Random Generator
**Acceptance criteria:**

Abstract class Range Random Generator should:
- have a function to return one integer from a specific range
- have a function to return pair of integers from a specific range

### PR DESCRIPTION
**Pull request by Rosein**
Closes #3

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,12 +18,14 @@ add_library(
...
   src/Minesweeper.cpp
   src/Gameboard.cpp
+  src/RandomGenerator.cpp
 )
 
 add_executable(
   minesweeper-uts
   tests/GameboardTest.cpp
   tests/MinesweeperTest.cpp
+  tests/RandomGeneratorTest.cpp
```
Comment (dariannax3) [inline CMakeLists.txt:28]:
```suggestion
  tests/RandomGeneratorsTest.cpp
```

File: inc/Gameboard.hpp [hunk 2]
```diff
@@ -1,11 +1,12 @@
 #pragma once
+
 #include <memory>
-#include "RandomGeneratorI.hpp"
 
-using RandomGeneratorPtr = std::shared_ptr<RandomGeneratorI>;
+#include "RandomGenerator.hpp"
 
+using RandomFromRangeGeneratorPtr = std::shared_ptr<RandomFromRangeGeneratorI>;
```
Comment (dariannax3) [inline inc/Gameboard.hpp:7]: I think better name is RandomRangeGenerator

**Review by dariannax3 — CHANGES_REQUESTED** [review]

File: inc/Gameboard.hpp (hunk 2 above)
Comment (Rosein) [reply inc/Gameboard.hpp:7]: Done

File: inc/RandomGenerator.hpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cassert>
+#include <cstdlib>
+#include <ctime>
+
+class RandomFromRangeGeneratorI {
```
[the remark this answers is not in the document]
Comment (Rosein) [reply inc/RandomGenerator.hpp:7]: I thought that random number must be extracted **from** range

File: inc/RandomGenerator.hpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  virtual int getRandom() = 0;
+  virtual ~RandomRangeGeneratorI() {}
+};
+
+class StdRandomRangeGenerator : public RandomRangeGeneratorI {
+ public:
+  StdRandomRangeGenerator(int lower_boundary, int upper_boundary);
+  int getRandom() override;
+
+ protected:
```
Comment (dariannax3) [inline inc/RandomGenerator.hpp:18]: No need to use protected - use private instead.
Add final specifier.

File: src/RandomGenerator.cpp [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#include "../inc/RandomGenerator.hpp"
+
+void StdRandomRangeGenerator::setSeed() {
+  static bool once = true;
+  if (once) std::srand(std::time(NULL));
```
Comment (dariannax3) [inline src/RandomGenerator.cpp:5]: Add a new line between statement and seed.

File: tests/RandomGeneratorTest.cpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#include <gmock/gmock.h>
+
+#include <vector>
+
+#include "../inc/RandomGenerator.hpp"
+
+TEST(RandomRangeGeneratorTest,
+     GivenRangeFromZeroToTenExpectRandomFromZeroToTen) {
+  StdRandomRangeGenerator random_generator(0, 10);
+
```
Comment (dariannax3) [inline tests/RandomGeneratorTest.cpp:10]: delete this line

**Review by dariannax3 — CHANGES_REQUESTED** [review]

**Pull request merged**